scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -158,268 +146,7 @@ trait AndroidAppModule extends AndroidModule {
-      .filter(ref =>
-        ref.path.ext == "jar" &&
-          ref != androidSdkModule().androidJarPath()
-      )
-      .flatMap(ref => {
-        val dest = Task.dest / ref.path.baseName
-        os.unzip(ref.path, dest)
-        val lookupPath = dest / "META-INF"
-        if (os.exists(lookupPath)) {
-          os.walk(lookupPath)
-            .filter(os.isFile)
-            .filterNot(f => isExcludedFromPackaging(f.relativeTo(lookupPath)))
-        } else {
-          Seq.empty[os.Path]
-        }
-      })
-      .map(PathRef(_))
-      .toSeq
-  }
-
-  /**
-   * Optimizes the APK using the `zipalign` tool for better performance.
-   *
-   * For more details on the zipalign tool, refer to:
-   * [[https://developer.android.com/tools/zipalign zipalign Documentation]]
-   */
-  def androidAlignedUnsignedApk: T[PathRef] = Task {
-    val alignedApk: os.Path = Task.dest / "app.aligned.apk"
-
-    os.call((
-      androidSdkModule().zipalignPath().path,
-      "-f",
-      "-p",
-      "4",
-      androidUnsignedApk().path,
-      alignedApk
-    ))
-
-    PathRef(alignedApk)
-  }
-
-  /**
-   * Generates the command-line arguments required for Android app signing.
-   *
-   * Uses the release keystore if release build type is set; otherwise, defaults to a generated debug keystore.
-   */
-  def androidSignKeyDetails: T[Seq[String]] = Task {
-
-    val keystorePass = {
-      if (androidIsDebug()) debugKeyStorePass else androidReleaseKeyStorePass().get
-    }
-    val keyAlias = {
-      if (androidIsDebug()) debugKeyAlias else androidReleaseKeyAlias().get
-    }
-    val keyPass = {
-      if (androidIsDebug()) debugKeyPass else androidReleaseKeyPass().get
-    }
-
-    Seq(
-      "--ks",
-      androidKeystore().path.toString,
-      "--ks-key-alias",
-      keyAlias,
-      "--ks-pass",
-      s"pass:$keystorePass",
-      "--key-pass",
-      s"pass:$keyPass"
-    )
-  }
-
-  /**
-   * Signs the APK using a keystore to generate a final, distributable APK.
-   *
-   * The signing step is mandatory to distribute Android applications. It adds a cryptographic
-   * signature to the APK, verifying its authenticity. This method uses the `apksigner` tool
-   * along with a keystore file to sign the APK.
-   *
-   * If no keystore is available, a new one is generated using the `keytool` utility.
-   *
-   * For more details on the apksigner tool, refer to:
-   * [[https://developer.android.com/tools/apksigner apksigner Documentation]]
-   */
-  def androidApk: T[PathRef] = Task {
-    val signedApk = Task.dest / "app.apk"
-
-    val signArgs = Seq(
-      androidSdkModule().apksignerPath().path.toString,
-      "sign",
-      "--in",
-      androidAlignedUnsignedApk().path.toString,
-      "--out",
-      signedApk.toString
-    ) ++ androidSignKeyDetails()
-
-    Task.log.info(s"Calling apksigner with arguments: ${signArgs.mkString(" ")}")
-
-    os.call(signArgs)
-
-    PathRef(signedApk)
+    androidReleaseLibsRClasses()
```

Comment:
this doesn't make much sense semantically, why would generatedSources point to a release (or debug) mode . Why do we need to specify Release or Debug mode for `androidLibsRClasses`?